worlds/undertale/__init__.py

```diff
@@ -46,47 +46,64 @@ class UndertaleWorld(World):
 
     def _get_undertale_data(self):
         return {
-            "world_seed": self.multiworld.per_slot_randoms[self.player].getrandbits(32),
+            "world_seed": self.random.getrandbits(32),
             "seed_name": self.multiworld.seed_name,
             "player_name": self.multiworld.get_player_name(self.player),
             "player_id": self.player,
             "client_version": self.required_client_version,
             "race": self.multiworld.is_race,
-            "route": self.multiworld.route_required[self.player].current_key,
-            "starting_area": self.multiworld.starting_area[self.player].current_key,
-            "temy_armor_include": bool(self.multiworld.temy_include[self.player].value),
-            "only_flakes": bool(self.multiworld.only_flakes[self.player].value),
-            "no_equips": bool(self.multiworld.no_equips[self.player].value),
-            "key_hunt": bool(self.multiworld.key_hunt[self.player].value),
-            "key_pieces": self.multiworld.key_pieces[self.player].value,
-            "rando_love": bool(self.multiworld.rando_love[self.player].value),
-            "rando_stats": bool(self.multiworld.rando_stats[self.player].value),
-            "prog_armor": bool(self.multiworld.prog_armor[self.player].value),
-            "prog_weapons": bool(self.multiworld.prog_weapons[self.player].value),
-            "rando_item_button": bool(self.multiworld.rando_item_button[self.player].value)
+            "route": self.options.route_required.current_key,
+            "starting_area": self.options.starting_area.current_key,
+            "temy_armor_include": bool(self.options.temy_include.value),
+            "only_flakes": bool(self.options.only_flakes.value),
+            "no_equips": bool(self.options.no_equips.value),
+            "key_hunt": bool(self.options.key_hunt.value),
+            "key_pieces": self.options.key_pieces.value,
+            "rando_love": bool(self.options.rando_love.value),
+            "rando_stats": bool(self.options.rando_stats.value),
+            "prog_armor": bool(self.options.prog_armor.value),
+            "prog_weapons": bool(self.options.prog_weapons.value),
+            "rando_item_button": bool(self.options.rando_item_button.value)
         }
 
+    def get_filler_item_name(self):
+        if self.options.route_required == "all_routes":
+            junk_pool = junk_weights_all.copy()
+        elif self.options.route_required == "genocide":
+            junk_pool = junk_weights_genocide.copy()
+        elif self.options.route_required == "neutral":
+            junk_pool = junk_weights_neutral.copy()
+        elif self.options.route_required == "pacifist":
+            junk_pool = junk_weights_pacifist.copy()
+        else:
+            junk_pool = junk_weights_all.copy()
+        if not self.options.only_flakes:
+            return self.random.choices(list(junk_pool.keys()), weights=list(junk_pool.values()))[0]
```

Comment:
Hmmm I'm pretty sure that at no point during code does `junk_pool` get modified, so I don't understand why you are making a `.copy()` of the respective junk weights